File: webapp/backend/src/main/java/com/computronica/webapp/model/Usuario.java
// src/main/java/com/computronica/webapp/model/Usuario.java
package com.computronica.webapp.model;

import com.google.cloud.Timestamp;

import lombok.Data;

@Data
public class Usuario {
    private String id;
    private String codigoInstitucional;
    private String sede;
    private String nombre;
    private String apellido;
    private String correoInstitucional;
    private String contrasena;
    private TipoUsuario tipo = TipoUsuario.estudiante;
    private boolean estado = true;
    private Timestamp createdAt;
    private Timestamp updatedAt;
}

File: CompuTronica/app/src/main/java/com/example/computronica/Model/Usuario.kt
package com.example.computronica.Model

import com.google.firebase.Timestamp
import com.google.firebase.firestore.ServerTimestamp

data class Usuario(
    var id: String = "",
    var codigoInstitucional: String = "",
    var sede: String = "",
    var nombre: String = "",
    var apellido: String = "",
    var correoInstitucional: String = "",
    var contrasena: String = "",
    var tipo: TipoUsuario = TipoUsuario.estudiante,
    var estado: Boolean = true,
    @ServerTimestamp var createdAt: Timestamp? = null,
    var updatedAt: Timestamp? = null
)

enum class TipoUsuario {
    estudiante,
    profesor,
    administrativo
}
